src/components/Address/index.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import ENS from 'ethereum-ens'
+
+function useDebounce(value, timeout) {
+  const [debounceValue, setDebounceValue] = useState(value)
+
+  useEffect(
+    function() {
+      const timer = setTimeout(function() {
+        setDebounceValue(value)
+      }, timeout)
+
+      return function() {
+        clearTimeout(timer)
+      }
+    },
+    [value]
+  )
+  return debounceValue
+}
+
+async function resolveENSName(value, provider) {
+  const ethereumRegEx = /^0x[a-fA-F0-9]{40}$/
+  const ens = new ENS(provider)
+  if (ethereumRegEx.test(value)) {
+    return await ens
+      .reverse(value)
+      .name()
+      .then(async result => {
+        const address = await ens.resolver(result).addr()
+        if (value !== address) {
+          return {
+            type: 'error',
+            data: ''
+          }
+        } else {
+          return {
+            type: 'success',
+            data: {
+              resolved: result,
+              address: address
+            }
+          }
+        }
+      })
+      .catch(err => {
+        return {
+          type: 'error',
+          data: ''
+        }
+      })
+  } else {
+    if (value.includes('.')) {
+      return ens
+        .resolver(value)
+        .addr()
+        .then(response => {
+          return {
+            type: 'success',
+            data: {
+              resolved: response,
+              address: response
+            }
+          }
+        })
+        .catch(err => {
+          return {
+            type: 'error',
```

Comment:
This type: 'error' doesn't seem to be getting used, can you just return an error message instead?

Reply:
Its here because of this https://github.com/ensdomains/ens-app/pull/335#discussion_r287260816